src/main/java/InputProcess.kt

```diff
@@ -0,0 +1,278 @@
+    private var isPrintIntoFile: Boolean = false
+
+    private var inputString: String = ""
+    private var outputFilePath: String = ""
+
+    private var inputFormat: Int = NO_FORMAT // Default: DEC for encode // BASE64 for decode
+    private var outputFormat: Int = NO_FORMAT //Default: BASE64 for encode // DEC for decode
+
+    private var inputArray: MutableList<String> = ArrayList()
+    private var outputArray: MutableList<String> = ArrayList()
+
+    fun inputArgumentsExecutor(args: ArrayList<String>): Boolean {
+        if(args.size == 1) return false
+        if (!inputArgumentsInspector(args)) return false
+
+        if (isToEncode) {
+            val bechData = BechTools.convertStringToBechData(inputString, inputFormat)
+            val encrypted = Encode.encode(bechData.humanReadablePart, bechData.data)
+            if (isPrintIntoFile)
+                FileInspector.writeStringIntoFile(outputFilePath, encrypted)
+            else
+                println(encrypted)
+        } else if (isToDecode) {
+            val decoded = Decode.decodingString(inputString)
+            if (isPrintIntoFile)
+                FileInspector.writeStringIntoFile(
+                    outputFilePath,
+                    BechTools.convertBechDataToString(decoded, outputFormat)
+                )
+            else
+                println(BechTools.convertBechDataToString(decoded, outputFormat))
+        }
+        return true
+    }
+
+    private fun inputArgumentsInspector(args: ArrayList<String>): Boolean { //Mikita
+        if (args.isEmpty()) {
+            UserCommunicator.printPrompt("You didn't provide any arguments :(")
+            return false
+        }
+
+        if (isArgIsHelp(args)) return false // if user ask for a help -> stop processing
+
+        divideArray(args) // divide array into input part and output part
+
+        if (!inputArrayInspector(inputArray)) return false
+        if (!outputArrayInspector(outputArray)) return false
+
+        return true
+    }
+
+    private fun divideArray(args: ArrayList<String>) {
+        inputArray = ArrayList()
+        outputArray = ArrayList()
+        var switch = false
+        args.forEachIndexed { _, value ->
+            if (value == "-o")
+                switch = true
+            if (!switch)
+                inputArray.add(value)
+            else
+                outputArray.add(value)
+        }
+    }
+
+    private fun inputArrayInspector(args: MutableList<String>): Boolean {
+        var isOperationSet = false
+        var isFormatSet = false
+        var isFileOutputSet = false
+        var isStringSet = false
+        args.forEachIndexed { _, value ->
+            when (value) {
+                "-e" -> {
+                    if (!isOperationSet) {
+                        isOperationSet = true
+                        isToEncode = true
+                    } else {
+                        println("Wrong input part of args: $value")
```

Comment:
Line println("Wrong input part of args: $value") is repeated many times and could be extracted as a separate method. It would be easier to change error message if you wanted to as it would be only at one place.